src/index.js

```diff
@@ -15,7 +15,14 @@ async function start() {
   const ec2InstanceId = await aws.startEc2Instance(label, githubRegistrationToken);
   setOutput(label, ec2InstanceId);
   await aws.waitForInstanceRunning(ec2InstanceId);
-  await gh.waitForRunnerRegistered(label);
+
+  try {
+    await gh.waitForRunnerRegistered(label);
+  }
```

Comment:
NIT, should we add waitForInstanceRunning to try-catch too? Sometimes AWS instances might fail to start and we should terminate them when timeout.
```suggestion
  try {
    await aws.waitForInstanceRunning(ec2InstanceId);
    await gh.waitForRunnerRegistered(label);
  }
```

Reply:
Good point, indeed we need to handle waitForInstanceRunning too.

Reply:
force pushed, addressed your comment in the latest commit. Many thanks